### ISSUE TITLE
Error when singularizing "radius"

### ISSUE THREAD

**Issue by dunglas** [doctrine/inflector#169]
`Inflector::singularize('radius')` returns `radiu` instead of keeping `radius`. (using v2).

**Comment by khalyomede** [comment]
I confirm, there might be a **missing unit test to add**, not to remove an -s to words that their singular form is equal to their plural.

Minimal reproducible snippet:

```php
use Doctrine\Inflector\InflectorFactory;

$inflector = InflectorFactory::create()->build();

echo $inflector->singularize("radius"); // "radiu"
echo $inflector->singularize("axis"); // "axi"
```

**Comment by garrettw** [comment]
Might be worth checking if a majority of "-us" words are in fact singular, and maybe can be pluralized to "-i".
Same for "axis" - check if a majority of "-is" words are singular and can be pluralized to "-es".

**Comment by Theremsoe** [comment]
Another word: ```Inflector::singularize('metadata')``` returns ```metadatum```.

Using v2

**Comment by RickKukiela** [comment]
I honestly do not know how this library is usable for singularization if it cannot even get something simple like virus or radius correct. How is this two years old and still a problem?

**Comment by malarzm** [comment]
> I honestly do not know how this library is usable for singularization if it cannot even get something simple like virus or radius correct. How is this two years old and still a problem?

@RickKukiela sending us a pull request with a fix would require just a little bit more of your precious time than useless complaining.

**Comment by RickKukiela** [comment]
@malarzm you're right I shouldn't complain. I just had a momentary lapse in judgement so I apologize for that. That being said, I just used my time to find another library that works better. I'm just sorry that the maintainers of this project don't seem to mind that it's not very good at one of its core features. If this was my project I would have definitely addressed this issue myself rather than waiting for a PR but that's just me. To each their own I guess.

I also want to point out that it seems you're the first individual from the "org" that has even commented on this topic in the two years it existed....

**Comment by malarzm** [comment]
> If this was my project I would have definitely addressed this issue myself rather than waiting for a PR but that's just me.

Please note that inflector is not the only package we maintain. Even if we, entire organization, would be working full time (we don't) we wouldn't have enough time to fix each and every issue that is reported. And here comes the beauty of open source software: anybody can help! Inflector is a really simple package comparing to ORM/DBAL/ODM and does not require specific knowledge other than language you're working with. 

> I also want to point out that it seems you're the first individual from the "org" that has even commented on this topic in the two years it existed....

It's not like the issue is requiring maintainer's input: it's straightforward and nothing to discuss really. "Just" needs somebody caring enough to add a rule and a test for it.

**Comment by RickKukiela** [comment]
Fair enough.

My point of view on this is pretty simple though. I'm working on a project I need an inflection library to generate singularizations. I install this library and test it and my _first_ test fails. Could I spend time doing a PR and adding a rule and a test? Sure. However, the fact that the first thing I tested failed instilled absolutely no confidence in me, that this would be a one time fix and that it would otherwise be good to go. It basically made me feel like this is going to be a problem - add to the fact I come here and find a 2 year old issue with the same complaint and no response or fix since. If I were using this library with good success for a while, and then randomly came across an issue like this, then I would be much more willing to do a PR and add fix/feature. Since I was still in "scouting" mode to try to find something for my needs, having to begin my experience with your library by taking time away from my project budget to add this feature/fix was not appealing to me at all which is why I just found another package that did not require me to jump through hoops for basic functionality.

We can have different points of view on this. I'm not really interested in arguing further. We have both spoken our peace on the  matter so lets just move on :)

**Comment by derrabus** [comment]
I'm amazed that people can spend more time on elaborating how unusable a free piece of software is than it would take to fix the actual problem they're complaining about.

**Comment by RickKukiela** [comment]
If you think it took me more time to type that than it would to fork this project, implement fixes, implement tests, verityify everything, push, and wait for a merge than it took me to just install something that works and type a few paragraphs then you must type really slow my guy.

**Comment by greg0ire** [comment]
> We can have different points of view on this.

Absolutely, it's just that your point of view is bad. I could waste more time going through the long list of reasons why that is, but I'm just going to lock this instead. Pretty sure deep down you know the reasons already anyway.
